skimage/feature/texture.py

```diff
@@ -136,6 +140,15 @@ def graycomatrix(image, distances, angles, levels=None, symmetric=False, normed=
     if levels is None:
         levels = 256
 
+    # apply mask if provided; masked pixels are set to maximum grayscale to
+    # place them outside of the computation range of `_glcm_loop`.
+    if mask is not None:
+        if mask.shape != image.shape or mask.dtype != np.bool_:
+            raise ValueError(
+                "The mask must have the same shape as the image and be of boolean type."
+            )
+        image[~mask] = np.iinfo(image.dtype).max
```

Comment:
I tried to come up with a minimal test case that covers an image where values are masked that corresponds to the data types maximum. 

```python
import numpy as np
import skimage as ski

image = np.array([[255, 255], [255, 255]], dtype=np.uint8)
mask = image > 1
result = ski.feature.graycomatrix(image, distances=[1], angles=[0], mask=mask)
assert result.max() == 0
```

I'd have thought, that this passes because all values are masked. But unfortunately the assert fails because `result[255, 255, 0, 0]` is actually `2`. 

I think this is due to a flaw in our optimization logic. The dtype's maximum is actually something that the `_glcm_loop` isn't skipping. E.g. in this case `levels` is 265 so 255 is counted. 

I'm not sure how to resolve this yet, but it looks like the optimization we had in mind, isn't possible because the full dtype range in `image` can be valid if `levels` is choosen with "dtype.max + 1"... :frowning_face:

Reply:
Basically if `image.max() == np.iinfo(image.dtype).max` we got a problem.

Reply:
Ouch. I can't believe I overlooked this. Good catch! Two things:
* If the entire range of the image datatype is used, then there is no value we can use to go outside the increment loop in `_glcm_loop()`; so I do agree that the approach is flawed. But...
* The output that you got in this example is expected. My interpretation of a mask is that a pixel is ignored if its mask value is False. Your interpretation seems to be the opposite. 

After looking at other common Python libraries, I noticed that your approach is more widely adopted, and I will consider this when making necessary corrections.